lib/hexagon/application.ex

```diff
@@ -8,6 +8,10 @@ defmodule Hexagon.Application do
   @impl true
   def start(_type, _args) do
     children = [
+      # Start default Finch HTTP pool
+      {Finch, name: Hexagon.FinchPool, pools: %{:default => [size: 32]}},
```

Comment:
Nitpick, take it or leave it: Since we have an atom can we use that syntax?
```suggestion
      {Finch, name: Hexagon.FinchPool, pools: %{default: [size: 32]}},
```

Not sure if it's needed or not but https://github.com/wojtekmach/req is built on top of Finch and has some neat features.

Reply:
I don't really have a strong feeling about http clients 🤷‍♂️

Reply:
tbh, after looking at it a bit more, I think I like Tesla + Finch as a solid http client that needs advanced things like retry and timeout mechanics.